src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -115,6 +115,16 @@ protected override void ApplyMissingCurrentAnalysisDataForUnhandledExceptionData
                 base.ApplyMissingCurrentAnalysisDataForUnhandledExceptionData(dataAtException.CoreAnalysisData, CurrentAnalysisData.CoreAnalysisData, throwBranchWithExceptionType);
             }
 
+            protected override TaintedDataAbstractValue GetDefaultValueForParameterOnEntry(IParameterSymbol parameter, AnalysisEntity analysisEntity)
+            {
+                if (this.DataFlowAnalysisContext.SourceInfos.IsSourceParameter(parameter, WellKnownTypeProvider))
+                {
+                    return TaintedDataAbstractValue.CreateTainted(parameter, parameter.DeclaringSyntaxReferences[0].GetSyntax(), this.OwningSymbol);
```

Comment:
`parameter.DeclaringSyntaxReferences[0].GetSyntax()` - terrible hack?

Reply:
@mavasani, do you have any better ideas here? We want to track the location of the origin of the tainted data (in this case, a method parameter).

Reply:
Given that you eventually need to store the location, I don't see a good alternative: https://github.com/dotnet/roslyn-analyzers/blob/8bc7b0bf842d3fee3ee8595e783828f8c4ee448e/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SymbolAccess.cs#L39-L42

For PointsToAbstractValue, we need similar special casing for parameters as there is no associated operation. We handle that by creating a special abstract location: https://github.com/dotnet/roslyn-analyzers/blob/8bc7b0bf842d3fee3ee8595e783828f8c4ee448e/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AbstractLocation.cs#L63-L64

Not sure you can do that here as you seem to really need to store the location in your symbol access type.

Reply:
I think this approach is fine for now.

If we run into weirdness with it, maybe we can do some special casing like:

1. In `GetDefaultValueForParameterOnEntry()`, we somehow mark the abstract value as needing its syntax location needs to be updated.
2. In `ComputeAnalysisValueForReferenceOperation()` (or somewhere), we see if it's an `IParameterReferenceOperation`, and if its current abstract value is still tainted and marked for its syntax location needs to be updated, return a new abstract value with the `IParameterReferenceOperation`'s syntax location.